styles/colors-silicon-night.less

```diff
@@ -11,8 +11,9 @@
 @agate-bg-color: black;
 @agate-bg-image: none;
 @agate-bg-image2: none;
-@agate-text-color: @agate-foreground;
-@agate-title-text-color: @agate-foreground;
+@agate-text-color: @agate-background;
+@agate-title-text-color: @agate-background;
+@agate-background: @agate-foreground;
```

Comment:
This doesn't seem right. I think it may make more sense to update these (foreground, background) general color assignments, rather than reverse where they are assigned.

If they're updated, many other values will automatically pick up the correct colors and work on other components will become simpler, since you're establishing the relationships between the variables and ultimately their color values.

```css
@agate-background: @silicon-black;
@agate-foreground: @silicon-white;
```

The `@silicon` colors are defined in the main Silicon skin colors file (day), which is the basis for this file. All variables defined there are accessible here.

Since the relationship between the conceptual variable `@agate-text-color` and the conceptual color `@agate-foreground` is already established in the other (day) file, you actually don't even need to recreate that rule here, even though the values are updated in this file. Each of these files can be considered to have their own scope.